backend/functions/lib/testing/analysis/acms-dxtr-divisions/compare-divisions.ts

```diff
@@ -0,0 +1,442 @@
+21,SV,Southern District of Georgia,343,Dublin
+21,SV,Southern District of Georgia,344,Savannah
+21,SV,Southern District of Georgia,345,Waycross
+21,SV,Southern District of Georgia,346,Statesboro
+21,TL,Northern District of Florida,291,Gainesville
+21,TL,Northern District of Florida,293,Pensacola
+21,TL,Northern District of Florida,294,Tallahassee
+21,TL,Northern District of Florida,295,Panama City
+21,TP,Middle District of Florida,308,Tampa`;
+
+type AcmsRegionDivision = {
+  acmsRegion: string;
+  acmsGroup: string;
+  acmsJudicialDistrictName: string;
+  acmsDivisionCode: string;
+  acmsDivisionName: string;
+};
+
+type DxtrRegionDivision = {
+  dxtrRegion: string;
+  dxtrRegionName: string;
+  dxtrGroup: string;
+  dxtrJudicialDistrictName: string;
+  dxtrDivisionCode: string;
+  dxtrDivisionName: string;
+  oktaGroupName: string;
+  camsOfficeCode: string;
+  ustpOfficeName: string;
+};
+
+type DivisionMatch = { acms: AcmsRegionDivision; dxtr: DxtrRegionDivision };
+type JoinedDivisions = AcmsRegionDivision & DxtrRegionDivision;
+
+type Output = {
+  match: DivisionMatch[];
+  noMatch: DivisionMatch[];
+  ignored: DivisionMatch[];
+};
+
+function loadAcmsRegionDivisions() {
+  const records = csv.split('\n');
+  const header = records.shift();
+  const keys = header.split(',') as (keyof AcmsRegionDivision)[];
+  const divisions = records.map((record) => {
+    const obj = {};
+    const values = record.split(',');
+    keys.forEach((key, idx) => {
+      obj[key] = values[idx];
+    });
+    return obj as AcmsRegionDivision;
+  });
+  return divisions;
+}
+
+export function ustpOfficeToDivision(ustp: UstpOfficeDetails): DxtrRegionDivision[] {
+  const courtDivisions: DxtrRegionDivision[] = [];
+  ustp.groups.reduce((acc, group) => {
+    group.divisions.forEach((division) => {
+      acc.push({
+        dxtrRegion: ustp.regionId,
+        dxtrRegionName: ustp.regionName,
+        dxtrGroup: group.groupDesignator,
+        dxtrJudicialDistrictName: division.court.courtName,
+        dxtrDivisionCode: division.divisionCode,
+        dxtrDivisionName: division.courtOffice.courtOfficeName,
+        oktaGroupName: ustp.idpGroupId,
+        camsOfficeCode: ustp.officeCode,
+        ustpOfficeName: ustp.officeName,
+      });
+    });
+    return acc;
+  }, courtDivisions);
+  return courtDivisions;
+}
+
+async function writeToCsv(records: JoinedDivisions[], fileName: string) {
+  if (records.length === 0) return;
+
+  const outputDirectory = './temp/acms-dxtr-divisions';
+  await fs.mkdir(outputDirectory, { recursive: true });
+  const file = await fs.open(`${outputDirectory}/${fileName}`, 'w');
+
+  file.write(Object.keys(records[0]).join(',') + '\n');
+
+  records.forEach((record) => {
+    const values: string[] = [];
+    Object.keys(record).forEach((key) => values.push(record[key]));
+    values.forEach((value, idx) => {
+      if (value.includes(',')) values[idx] = '"' + value + '"';
+    });
+    file.write(values.join(',') + '\n');
+  });
```

Comment:
I assume line 339 is writing the CSV header line.  Does the loop starting at 341 write the header a second time?

Doesn't line 343 do the same thing as Object.values(record). ??

On 345 I'm not sure I understand why we're looking to see if value includes ',' and only then adding the value to values[].